lib/gracenote/search.rb

```diff
@@ -2,14 +2,11 @@ module Gracenote
   module Search
     # @params mode [String] 'SINGLE_BEST' or 'SINGLE_BEST_COVER'
     def search(artist: nil, album: nil, track: nil, mode: nil)
-      params = {}
-      params[:mode] = mode if mode
-      params[:text] = { :@TYPE => 'ARTIST', :content! => artist } if artist
-      params[:text] = { :@TYPE => 'ALBUM_TITLE', :content! => album } if album
-      params[:text] = { :@TYPE => 'TRACK_TITLE', :content! => track } if track
-
-      query("album_search", params, auth: true, lang: true) do |response|
-        response
+      query("album_search", auth: auth, lang: lang) do |xml|
+        xml.MODE { xml.text mode } if mode
+        xml.TEXT(TYPE: 'ARTIST') { xml.text artist } if artist
```

Comment:
__[rubocop]__
* Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)